stratosphere/pm/source/boot2/boot2_api.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2018-2019 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cctype>
+#include <dirent.h>
+#include <stratosphere/cfg.hpp>
+#include <stratosphere/ldr.hpp>
+#include <stratosphere/pm.hpp>
+#include "boot2_api.hpp"
+
+namespace sts::boot2 {
+
+    namespace {
+
+        /* Launch lists. */
+
+        /* psc, bus, pcv is the minimal set of required titles to get SD card. */
+        /* bus depends on pcie, and pcv depends on settings. */
+        constexpr ncm::TitleId PreSdCardLaunchPrograms[] = {
+            ncm::TitleId{TitleId_Psc},         /* psc */
```

Comment:
Should there be an implicit constructor in `ncm::TitleId` that accepts an u64?

Reply:
It's important that std::is_pod<>::value be true. I think that's incompatible with constructor?

I should make all the TitleId_* definitions be ncm::TitleId though.